src/gsy_e/models/strategy/energy_parameters/heat_pump_tank.py

```diff
@@ -0,0 +1,311 @@
+        return TankSolverParameters(
+            tank_volume_L=self._parameters.tank_volume_L,
+            current_storage_temp_C=current_storage_temp_C,
+            target_storage_temp_C=target_storage_temp_C,
+        )
+
+    def create_tank_parameters_without_target_tank_temp(self, time_slot: DateTime):
+        """
+        Create tank parameters without specifying tank temp, in order to be used as an output of
+        the solver.
+        """
+        current_storage_temp_C = self._state.get_storage_temp_C(time_slot)
+        return TankSolverParameters(
+            tank_volume_L=self._parameters.tank_volume_L,
+            current_storage_temp_C=current_storage_temp_C,
+        )
+
+    def _temp_diff_to_Q_kWh(self, diff_temp_K: float) -> float:
+        return diff_temp_K * self._Q_specific
+
+    def _Q_kWh_to_temp_diff(self, energy_kWh: float) -> float:
+        return energy_kWh / self._Q_specific
+
+
+class AllTanksEnergyParameters:
+    """Manage the operation of heating and extracting temperature from multiple tanks."""
+
+    def __init__(self, tank_parameters: List[TankParameters]):
+        self._tanks_energy_parameters = [
+            TankEnergyParameters(tank, GlobalConfig.slot_length) for tank in tank_parameters
+        ]
+
+    def increase_tanks_temp_from_heat_energy(self, heat_energy: float, time_slot: DateTime):
+        """Increase the temperature of the water tanks with the provided heat energy."""
+        # Split heat energy equally across tanks
+        heat_energy_per_tank = heat_energy / len(self._tanks_energy_parameters)
+        for tank in self._tanks_energy_parameters:
+            tank.increase_tank_temp_from_heat_energy(heat_energy_per_tank, time_slot)
+
+    def decrease_tanks_temp_from_heat_energy(self, heat_energy: float, time_slot: DateTime):
+        """Decrease the temperature of the water tanks with the provided heat energy."""
+        heat_energy_per_tank = heat_energy / len(self._tanks_energy_parameters)
+        for tank in self._tanks_energy_parameters:
+            tank.decrease_tank_temp_from_heat_energy(heat_energy_per_tank, time_slot)
+
+    def set_temp_decrease_vhp(
+        self, heat_energy: float, time_slot: DateTime
+    ) -> List[TankSolverParameters]:
+        """
+        Decrease the temperature of all tanks according to the heat energy produced.
+        Returns true if there is unmatched heat demand in any of the tanks.
+        """
+        heat_energy_per_tank = heat_energy / len(self._tanks_energy_parameters)
+        unmatched_heat_demand = [
+            tank.decrease_tank_temp_vhp(heat_energy_per_tank, time_slot)
+            for tank in self._tanks_energy_parameters
+        ]
+        # TODO: Implemented the current behavior, which is wrong because setting the current tank
+        # temp to target tank temp ignores the energy that is already consumed from the heatpump
+        # and the corresponding temp increase.
```

Comment:
Isn't "energy that is already consumed from the heatpump" dealt with [later on](https://github.com/gridsingularity/gsy-e/pull/1775/files#diff-2f2237c0fd6b6d25a18ddfe4800b4eec7b658c686d250fcbd9404cab7764beb4R176) ?
IMO in this method here we should not deal with the increase of the storage temp because of the name of the method.

Reply:
I was wrong when I left this comment, I checked again and indeed the consumed energy is dealt with in the calculation of the unmatched heat demand. Also:
> we should not deal with the increase of the storage temp
this method does not handle that, it only saves the decrease of the temp, and reports if the temp decrease brings the water tank at a state lower than the min temp. So I think that we agree on all, deleting the comment and thanks for pointing it out!